buildtest/utils/command.py

```diff
@@ -1,7 +1,69 @@
+from buildtest.utils.file import read_file
+import os
 import locale
 import subprocess
 import shlex
 import shutil
+import tempfile
+
+
+class Capturing:
+    """capture output from stdout and stderr into capture object.
+       This is based off of github.com/vsoch/gridtest but modified
+       to write files. The stderr and stdout are set to temporary files at
+       the init of the capture, and then they are closed when we exit. This
+       means expected usage looks like:
+
+       with Capturing() as capture:
+           process = subprocess.POpen(...)
+           
+
+       And then the output and error are retrieved from reading the files:
+       and exposed as properties to the client:
+
+           capture.out
+           capture.err
+
+       And cleanup means deleting these files, if they exist.
+    """
+
+    def __enter__(self):
+        self.set_stdout()
+        self.set_stderr()
+        return self
+
+    def set_stdout(self):
+        self.stdout = open(tempfile.mkstemp()[1], "w")
+
+    def set_stderr(self):
+        self.stderr = open(tempfile.mkstemp()[1], "w")
+
+    def __exit__(self, *args):
+        self.stderr.close()
```

Comment:
I don't think it's possible to run into issue closing file handle but does it make sense to put an exception let's say with **try except**. Only thing i can think of is if file is read and deleted in time before file is closed it could result in an exception.

Reply:
This wouldn't be an error (I'm not sure you tested this?) there isn't any error if the file is closed (and then is closed again)
```
with open('/tmp/test.x', 'w') as filey: 
    print(filey) 

<_io.TextIOWrapper name='/tmp/test.x' mode='w' encoding='UTF-8'>

filey.close()                                                                                        
filey.close()   
```

OR if the file is deleted (and then closed again)

```python
import os
os.remove(filey.name)                                                                               
filey.name                                                                                          
 '/tmp/test.x'
filey.close()                                                                                       
```
Tada, no errors thrown! Good thought though. :)

Reply:
Nope i didn't test i was just thinking of an edge case where we run into an exception. Thanks for the test, it's good to know we close file-handle gracefully in the event if file is not present.